### ISSUE TITLE
BUG - Sev-1 - iOS (v 17.3) - Can't navigate most pages using keyboard

### ISSUE THREAD

**Issue by rbontrager** [department-of-veterans-affairs/va-mobile-app#9288]
## What happened?

You can no longer navigate most pages using the keyboard on iOS running 17.5.  This includes pages like Claims, ContactInfo etc.  I confirmed that you can navigate these pages on version 16.3.

## Specs:

- Device: iOS
- OS: 17.5
- User Account:
- Accessibility State: Keyboard

## Steps to Reproduce

1. Navigate to Contact Info. Notices you can't get the keyboard to go past 'How we use your contact information'
2. Navigate to Claims. Notice you can't get the keyboard to go past the 'Claims' back button
3. Navigate to other places in the app and notices that not everything is navigable using the keyboard 

If you haven't navigated the app with a keyboard before - connect any bluetooth keyboard to your device. Then in your device's settings, go to Accessibility, then Keyboards (it's in the physical and motor section), and toggle "Full Keyboard Access" on. You should be able to start using your keyboard at this point - and common commands are helpfully listed onscreen (feel free to remap in the commands section if desired):
<img width="503" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBNHRsQmc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--e5bfe2d98b428093d736007201038d029213be75/image.png" alt="image.png" />

## Acceptance Criteria

Able to navigate via keyboard through the whole app, even in latest iOS versions

## Bug Severity - BE SURE TO ADD THE SEVERITY LABEL
 
See [Bug Tracking](https://department-of-veterans-affairs.github.io/va-mobile-app/docs/QA#issue-severity) for details on severity levels

- Impact: High Low 
- Frequency: High Low 

- 3 - Low
- 2 - High
- 1 - Critical

## Screen shot(s) and additional information

<details>
  <summary>Full JSON response for services related to issue (expand/collapse)</summary>
  

</details>

[Private Zenhub Video](https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBNE5aQmc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--51f5e10117986cd491339fe9341d70e0c5fa061b/RPReplay_Final1723053074.mp4)

## Ticket Checklist
- [ ] Steps to reproduce are defined
- [ ] Desired behavior is added
- [ ] Labels added (front-end, back-end, global, Health, relevant feature, accessibility, etc)
- [ ] Estimate of 1 added (for front-end tickets)
- [ ] Added either to the relevant feature epic (if found during new feature implementation), or the relevant team's bug epic ([Global](https://app.zenhub.com/workspaces/va-mobile-60f1a34998bc75000f2a489f/issues/gh/department-of-veterans-affairs/va-mobile-app/8302), [Health & Benefits](https://app.zenhub.com/workspaces/va-mobile-60f1a34998bc75000f2a489f/issues/gh/department-of-veterans-affairs/va-mobile-app/8303), [API](https://app.zenhub.com/workspaces/va-mobile-60f1a34998bc75000f2a489f/issues/gh/department-of-veterans-affairs/va-mobile-app/8304), [QART](https://app.zenhub.com/workspaces/va-mobile-60f1a34998bc75000f2a489f/issues/gh/department-of-veterans-affairs/va-mobile-app/8305))

**Comment by TKDickson** [comment]
Upgrading this to a sev-1 bug, and escalating accordingly.

**Comment by TKDickson** [comment]
Reproduced today on a device I just upgraded to 17.6.1, with today's develop build.

**Comment by TKDickson** [comment]
Changed title - I connected to a 17.3 device (iPad mini) and had the same navigation issues.

**Comment by cadibemma** [comment]
Ticket update in [Slack](https://adhoc.slack.com/archives/C02N65QRRB7/p1724271855606309)

**Comment by TKDickson** [comment]
Meeting notes from today (@brea11y @dumathane @cadibemma and me):
* This is caused by an interaction of (some newer version of React Native) and iOS 17.X versions. Seems to have impacted the mobile app for somewhere in the realm of ~1-2 months. We're not digging too deeply into this right now because we are focused on getting a fix out.
* We do not believe we have any user reports (feedback hub, direct avenues like discussions with BVA, etc), unless it's the same root cause as bug ticket #9442 .
* This [patch](https://github.com/facebook/react-native/issues/43266#issuecomment-2176151625) (and iterating on it a few different ways in our code) did not work for us to solve the problem.
* We considered, but decided against preliminarily, options like using Availability Framework to notify all iOS users of keyboard navigation problems.
* The approach we're taking now is to deliver a change in the 2.36 release of using the 'button' or 'link' accessibility roles for as many of the problematic components as possible. For some components, this is a change that our accessibility specialist wanted anyways (in the long run); for other components, we'll be using the accessibility label when possible to give additional context when using a screen reader (ex: checkboxes likely mapped to a role of button, with a label of checkbox).
* There are hundreds of instances in the code that might need remapping to function again with a keyboard. We're going to focus on high frequency screens and flows (as defined in our [bug severity process](https://department-of-veterans-affairs.github.io/va-mobile-app/docs/QA/QualityAssuranceProcess/#frequency-definitions)), to restore as much functionality as possible while still delivering something this sprint.

**Comment by TKDickson** [comment]
**Test Plan**
Smoke testing in demo mode that I can:
* navigate into the list details screen for all of the features in the app (ex: vaccine details, payment details, upcoming appt details, etc), and
* send a new message

In both iOS and Android, with all (basic) assistive tech:
* Screen reader
* Keyboard only (focus of this fix)
* Voice control
* Switch control
* Text resizing

Known/things found during 'side-by-side' fix this morning (Eng, QA and accessibility specialist) that I'm writing up additional tickets for:
* "Actionable offscreen" inaccessible to keyboard only users
* Screen reader doesn't announce what's currently selected in picker (ex: last 3 months for past appts)
* We should spend some more time investigating whether a modification of the RN patch would work for us
* When this is eventually resolved/we don't have to use this temporary fix, there are some (but not all) roles we changed here that we'd want to change back

**Comment by TKDickson** [comment]
Found during testing: Android keyboard gets trapped behind layers of the UI (large panels, native popups)

**Comment by TKDickson** [comment]
Wrote #9474 and #9476, coming out of this morning's conversation about what's possible to include in the fix for this ticket.

**Comment by TKDickson** [comment]
Doesn't require further testing in develop; closing
